vcd/resource_vcd_vapp_vm.go

```diff
@@ -1513,21 +1653,17 @@ func genericVcdVmRead(d *schema.ResourceData, meta interface{}, origin string, v
 	dSet(d, "cpu_hot_add_enabled", vm.VM.VMCapabilities.CPUHotAddEnabled)
 	dSet(d, "memory_hot_add_enabled", vm.VM.VMCapabilities.MemoryHotAddEnabled)
 
-	cpus := int64(0)
-	coresPerSocket := 0
-	memory := int64(0)
-	for _, item := range vm.VM.VirtualHardwareSection.Item {
-		if item.ResourceType == 3 {
-			cpus += item.VirtualQuantity
-			coresPerSocket = item.CoresPerSocket
-		}
-		if item.ResourceType == 4 {
-			memory = item.VirtualQuantity
-		}
-	}
-	dSet(d, "memory", memory)
-	dSet(d, "cpus", cpus)
-	dSet(d, "cpu_cores", coresPerSocket)
+	dSet(d, "memory", vm.VM.VmSpecSection.MemoryResourceMb.Configured)
```

Comment:
Any chance to have `nil` for `vm.VM.VmSpecSection.MemoryResourceMb` or `vm.VM.VmSpecSection.CpuResourceMhz`

Reply:
I haven't faced this in any condition, but you know there is always a possibility as API isn't ours.

Reply:
added just to be 100% safe